Packages/ca.tekly.common/Runtime/Utils/UnityRuntimeEditorUtils.cs

```diff
@@ -29,18 +46,24 @@ private static void PlayModeStateChanged(UnityEditor.PlayModeStateChange change)
             }
             
             UnityEditor.EditorApplication.delayCall += () => {
+                s_currentlyRunning = false;
                 UnityEditor.EditorApplication.playModeStateChanged -= PlayModeStateChanged;
-                foreach (var exitAction in s_onExitActions) {
-                    try {
-                        exitAction();
-                    } catch (Exception exception) {
-                        Debug.LogException(exception);
-                    }
-                }
-            
-                s_onExitActions.Clear();
+                RunAllTheThings(s_onExitActions);
             };
         }
+
+        private static void RunAllTheThings(List<Action> allTheThings)
+        {
+            foreach (var thing in allTheThings) {
+                try {
+                    thing();
+                } catch (Exception exception) {
+                    Debug.LogException(exception);
+                }
+            }
+
+            s_onExitActions.Clear();
```

Comment:
Seems like this `clear` should not be in `RunAllTheThings` but where this is called